### ISSUE TITLE
[camera] camera warning: A resource failed to call release

### ISSUE THREAD

**Issue by wwwqyhme** [flutter/flutter#108941]
I'm following [https://docs.flutter.dev/cookbook/plugins/picture-using-camera](https://docs.flutter.dev/cookbook/plugins/picture-using-camera) to write my camera application. but i got `A resource failed to call release` warnings sometimes after I restarted my appliacation in debug mode

<details>
<summary>console log </summary>

``` 
Restarted application in 858ms.
W/libc    (15949): Access denied finding property "vendor.camera.aux.packagelist"  
I/Camera  (15949): close  
D/CameraExtImplXiaoMi(15949): releaseCameraDevice: 0  
D/CameraDevice-JV-0(15949): close: E. id = 0  
W/CameraDevice-JV-0(15949): Device error received, code 4, frame number 261, request ID 0, subseq ID 0  
W/libc    (15949): Access denied finding property "vendor.camera.aux.packagelist"  
W/CameraManagerGlobal(15949): ignore the torch status update of camera: 2  
W/libc    (15949): Access denied finding property "vendor.camera.aux.packagelist"  
W/CameraManagerGlobal(15949): ignore the torch status update of camera: 3  
W/libc    (15949): Access denied finding property "vendor.camera.aux.packagelist"  
W/CameraManagerGlobal(15949): ignore the torch status update of camera: 4  
W/libc    (15949): Access denied finding property "vendor.camera.aux.packagelist"  
W/CameraManagerGlobal(15949): ignore the torch status update of camera: 5  
D/CameraDevice-JV-0(15949): close: X  
E/CameraManagerGlobal(15949): Camera 4 is not available. Ignore physical camera status change  
I/Camera  (15949): open | onClosed  
W/CameraDevice-JV-0(15949): Device error received, code 4, frame number 259, request ID 0, subseq ID 0  
W/CameraDevice-JV-0(15949): Device error received, code 5, frame number 259, request ID 0, subseq ID 0  
W/CameraDevice-JV-0(15949): Device error received, code 5, frame number 260, request ID 0, subseq ID 0  
W/CameraDevice-JV-0(15949): Device error received, code 4, frame number 260, request ID 0, subseq ID 0  
W/CameraDevice-JV-0(15949): Device error received, code 5, frame number 261, request ID 0, subseq ID 0
W/CameraDevice-JV-0(15949): Device error received, code 4, frame number 262, request ID 0, subseq ID 0
W/CameraDevice-JV-0(15949): Device error received, code 5, frame number 262, request ID 0, subseq ID 0
W/CameraDevice-JV-0(15949): Device error received, code 4, frame number 263, request ID 0, subseq ID 0
W/CameraDevice-JV-0(15949): Device error received, code 5, frame number 263, request ID 0, subseq ID 0
W/CameraDevice-JV-0(15949): Device error received, code 3, frame number 264, request ID 0, subseq ID 0
W/Camera  (15949): The selected imageFormatGroup is not supported by Android. Defaulting to yuv420
W/libc    (15949): Access denied finding property "persist.vendor.camera.privapp.list"
D/CameraExtImplXiaoMi(15949): initCameraDevice: 0
E/CameraManagerGlobal(15949): Camera 4 is not available. Ignore physical camera status change
W/libc    (15949): Access denied finding property "vendor.camera.aux.packagelist"
W/CameraManagerGlobal(15949): ignore the torch status update of camera: 2
W/libc    (15949): Access denied finding property "vendor.camera.aux.packagelist"
W/CameraManagerGlobal(15949): ignore the torch status update of camera: 3
W/libc    (15949): Access denied finding property "vendor.camera.aux.packagelist"
W/CameraManagerGlobal(15949): ignore the torch status update of camera: 4
W/libc    (15949): Access denied finding property "vendor.camera.aux.packagelist"
W/CameraManagerGlobal(15949): ignore the torch status update of camera: 5
I/Camera  (15949): startPreview
D/CameraDevice-JV-0(15949): waitUntilIdle: E. id = 0
D/CameraDevice-JV-0(15949): waitUntilIdle: X
I/Camera  (15949): CameraCaptureSession onConfigured
I/Camera  (15949): Updating builder settings
D/Camera  (15949): Updating builder with feature: ExposureLockFeature
D/Camera  (15949): Updating builder with feature: ExposurePointFeature
D/Camera  (15949): Updating builder with feature: ZoomLevelFeature
D/Camera  (15949): Updating builder with feature: AutoFocusFeature
D/Camera  (15949): Updating builder with feature: NoiseReductionFeature
I/Camera  (15949): updateNoiseReduction | currentSetting: fast
D/Camera  (15949): Updating builder with feature: FocusPointFeature
D/Camera  (15949): Updating builder with feature: ResolutionFeature
D/Camera  (15949): Updating builder with feature: SensorOrientationFeature
D/Camera  (15949): Updating builder with feature: FlashFeature
D/Camera  (15949): Updating builder with feature: ExposureOffsetFeature
D/Camera  (15949): Updating builder with feature: FpsRangeFeature
I/Camera  (15949): refreshPreviewCaptureSession
I/Camera  (15949): CameraCaptureSession onClosed
W/System  (15949): A resource failed to call release.
```

</details>

<details>
<summary>source code</summary>

```  dart
import 'dart:async';
import 'dart:io';

import 'package:camera/camera.dart';
import 'package:flutter/material.dart';

Future<void> main() async {
  // Ensure that plugin services are initialized so that `availableCameras()`
  // can be called before `runApp()`
  WidgetsFlutterBinding.ensureInitialized();

  // Obtain a list of the available cameras on the device.
  final cameras = await availableCameras();

  // Get a specific camera from the list of available cameras.
  final firstCamera = cameras.first;

  runApp(
    MaterialApp(
      theme: ThemeData.dark(),
      home: TakePictureScreen(
        // Pass the appropriate camera to the TakePictureScreen widget.
        camera: firstCamera,
      ),
    ),
  );
}

// A screen that allows users to take a picture using a given camera.
class TakePictureScreen extends StatefulWidget {
  const TakePictureScreen({
    super.key,
    required this.camera,
  });

  final CameraDescription camera;

  @override
  TakePictureScreenState createState() => TakePictureScreenState();
}

class TakePictureScreenState extends State<TakePictureScreen> {
  late CameraController _controller;
  late Future<void> _initializeControllerFuture;

  @override
  void initState() {
    super.initState();
    // To display the current output from the Camera,
    // create a CameraController.
    _controller = CameraController(
      // Get a specific camera from the list of available cameras.
      widget.camera,
      // Define the resolution to use.
      ResolutionPreset.medium,
    );

    // Next, initialize the controller. This returns a Future.
    _initializeControllerFuture = _controller.initialize();
  }

  @override
  void dispose() {
    // Dispose of the controller when the widget is disposed.
    _controller.dispose();
    super.dispose();
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(title: const Text('Take a picture')),
      // You must wait until the controller is initialized before displaying the
      // camera preview. Use a FutureBuilder to display a loading spinner until the
      // controller has finished initializing.
      body: FutureBuilder<void>(
        future: _initializeControllerFuture,
        builder: (context, snapshot) {
          if (snapshot.connectionState == ConnectionState.done) {
            // If the Future is complete, display the preview.
            return CameraPreview(_controller);
          } else {
            // Otherwise, display a loading indicator.
            return const Center(child: CircularProgressIndicator());
          }
        },
      ),
      floatingActionButton: FloatingActionButton(
        // Provide an onPressed callback.
        onPressed: () async {
          // Take the Picture in a try / catch block. If anything goes wrong,
          // catch the error.
          try {
            // Ensure that the camera is initialized.
            await _initializeControllerFuture;

            // Attempt to take a picture and get the file `image`
            // where it was saved.
            final image = await _controller.takePicture();

            if (!mounted) return;

            // If the picture was taken, display it on a new screen.
            await Navigator.of(context).push(
              MaterialPageRoute(
                builder: (context) => DisplayPictureScreen(
                  // Pass the automatically generated path to
                  // the DisplayPictureScreen widget.
                  imagePath: image.path,
                ),
              ),
            );
          } catch (e) {
            // If an error occurs, log the error to the console.
            print(e);
          }
        },
        child: const Icon(Icons.camera_alt),
      ),
    );
  }
}

// A widget that displays the picture taken by the user.
class DisplayPictureScreen extends StatelessWidget {
  final String imagePath;

  const DisplayPictureScreen({super.key, required this.imagePath});

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(title: const Text('Display the Picture')),
      // The image is stored as a file on the device. Use the `Image.file`
      // constructor with the given path to display the image.
      body: Image.file(File(imagePath)),
    );
  }
}
```
</details>

<details>
<summary>flutter doctor </summary>

``` 
[√] Flutter (Channel stable, 3.0.5, on Microsoft Windows [版本 10.0.22000.795], locale zh-CN)
    • Flutter version 3.0.5 at C:\flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision f1875d570e (3 weeks ago), 2022-07-13 11:24:16 -0700
    • Engine revision e85ea0e79c
    • Dart version 2.17.6
    • DevTools version 2.12.2

[√] Android toolchain - develop for Android devices (Android SDK version 32.1.0-rc1)
    • Android SDK at C:\Users\wwwqyhme\AppData\Local\Android\sdk
    • Platform android-33, build-tools 32.1.0-rc1
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 11.0.11+9-b60-7590822)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[√] Visual Studio - develop for Windows (Visual Studio Community 2022 17.1.3)
    • Visual Studio at C:\Program Files\Microsoft Visual Studio\2022\Community
    • Visual Studio Community 2022 version 17.1.32328.378
    • Windows 10 SDK version 10.0.22000.0

[√] Android Studio (version 2021.1)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.11+9-b60-7590822)

[√] VS Code (version 1.69.2)
    • VS Code at C:\Users\wwwqyhme\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.46.0

[√] Connected device (4 available)
    • SHARK PAR A0 (mobile) • 824ef9ce • android-arm64  • Android 12 (API 31)
    • Windows (desktop)     • windows  • windows-x64    • Microsoft Windows [版本 10.0.22000.795]
    • Chrome (web)          • chrome   • web-javascript • Google Chrome 103.0.5060.134
    • Edge (web)            • edge     • web-javascript • Microsoft Edge 103.0.1264.77

HTTP Host availability check is taking a long time...[√] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
```

</details>

**Comment by maheshmnj** [comment]
Hi @wwwqyhme, Thanks for filing the issue. I ran your code sample and I am unable to reproduce the issue. I verified on Xiaomi k20 pro using the latest version of the camera plugin `camera: ^0.10.0`. Are there any known steps to reproduce the issue? What device is this issue on?

 
<details>
<summary>logs</summary>

```
Launching lib/main.dart on Redmi K20 Pro in debug mode...
Note: /Users/mahesh/Documents/flutter/.pub-cache/hosted/pub.dartlang.org/webview_flutter_android-2.9.2/android/src/main/java/io/flutter/plugins/webviewflutter/WebViewHostApiImpl.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.

✓  Built build/app/outputs/flutter-apk/app-debug.apk.
Connecting to VM Service at ws://127.0.0.1:49832/3g3wahqOjB0=/ws
I/CameraManagerGlobal(17807): Connecting to camera service
W/CameraManagerGlobal(17807): [soar.cts] ignore the status update of camera: 100
W/CameraManagerGlobal(17807): [soar.cts] ignore the status update of camera: 101
W/CameraManagerGlobal(17807): [soar.cts] ignore the status update of camera: 120
W/CameraManagerGlobal(17807): [soar.cts] ignore the status update of camera: 20
W/CameraManagerGlobal(17807): [soar.cts] ignore the status update of camera: 21
W/CameraManagerGlobal(17807): [soar.cts] ignore the status update of camera: 60
W/CameraManagerGlobal(17807): [soar.cts] ignore the status update of camera: 61
W/CameraManagerGlobal(17807): [soar.cts] ignore the status update of camera: 62
W/CameraManagerGlobal(17807): [soar.cts] ignore the status update of camera: 63
W/CameraManagerGlobal(17807): ignore the torch status update of camera: 100
W/CameraManagerGlobal(17807): ignore the torch status update of camera: 120
W/CameraManagerGlobal(17807): ignore the torch status update of camera: 20
W/CameraManagerGlobal(17807): ignore the torch status update of camera: 21
W/CameraManagerGlobal(17807): ignore the torch status update of camera: 60
W/CameraManagerGlobal(17807): ignore the torch status update of camera: 61
W/CameraManagerGlobal(17807): ignore the torch status update of camera: 62
W/CameraManagerGlobal(17807): ignore the torch status update of camera: 63
W/Camera  (17807): The selected imageFormatGroup is not supported by Android. Defaulting to yuv420
E/libc    (17807): Access denied finding property "persist.vendor.camera.privapp.list"
W/.example.sample(17807): type=1400 audit(0.0:382029): avc: denied { read } for name="u:object_r:vendor_persist_camera_prop:s0" dev="tmpfs" ino=19085 scontext=u:r:untrusted_app:s0:c105,c260,c512,c768 tcontext=u:object_r:vendor_persist_camera_prop:s0 tclass=file permissive=0 app=com.example.sample
I/Camera  (17807): startPreview
W/BpBinder(17807): PerfMonitor binderTransact :  time=396ms interface=android.hardware.camera2.ICameraDeviceUser code=6
I/Camera  (17807): CameraCaptureSession onConfigured
I/Camera  (17807): Updating builder settings
D/Camera  (17807): Updating builder with feature: ExposureLockFeature
D/Camera  (17807): Updating builder with feature: ExposurePointFeature
D/Camera  (17807): Updating builder with feature: ZoomLevelFeature
D/Camera  (17807): Updating builder with feature: AutoFocusFeature
D/Camera  (17807): Updating builder with feature: NoiseReductionFeature
I/Camera  (17807): updateNoiseReduction | currentSetting: fast
D/Camera  (17807): Updating builder with feature: FocusPointFeature
D/Camera  (17807): Updating builder with feature: ResolutionFeature
D/Camera  (17807): Updating builder with feature: SensorOrientationFeature
D/Camera  (17807): Updating builder with feature: FlashFeature
D/Camera  (17807): Updating builder with feature: ExposureOffsetFeature
D/Camera  (17807): Updating builder with feature: FpsRangeFeature
I/Camera  (17807): refreshPreviewCaptureSession
I/.example.sampl(17807): ProcessProfilingInfo new_methods=1426 is saved saved_to_disk=1 resolve_classes_delay=8000
D/MIUIInput(17807): [MotionEvent] ViewRootImpl { action=ACTION_DOWN, id[0]=0, pointerCount=1, eventTime=84240979, downTime=84240979 } moveCount:0
W/MirrorManager(17807): this model don't Support
D/MIUIInput(17807): [MotionEvent] ViewRootImpl { action=ACTION_UP, id[0]=0, pointerCount=1, eventTime=84241105, downTime=84240979 } moveCount:0
I/Camera  (17807): runPictureAutoFocus
I/Camera  (17807): lockAutoFocus
D/CameraCaptureCallback(17807): CameraCaptureCallback | state: STATE_WAITING_FOCUS | afState: 2 | aeState: 2
D/CameraCaptureCallback(17807): CameraCaptureCallback | state: STATE_WAITING_FOCUS | afState: 2 | aeState: 2
D/CameraCaptureCallback(17807): CameraCaptureCallback | state: STATE_WAITING_FOCUS | afState: 2 | aeState: 2
D/CameraCaptureCallback(17807): CameraCaptureCallback | state: STATE_WAITING_FOCUS | afState: 2 | aeState: 2
D/CameraCaptureCallback(17807): CameraCaptureCallback | state: STATE_WAITING_FOCUS | afState: 2 | aeState: 2
D/CameraCaptureCallback(17807): CameraCaptureCallback | state: STATE_WAITING_FOCUS | afState: 2 | aeState: 2
D/CameraCaptureCallback(17807): CameraCaptureCallback | state: STATE_WAITING_FOCUS | afState: 2 | aeState: 2
D/CameraCaptureCallback(17807): CameraCaptureCallback | state: STATE_WAITING_FOCUS | afState: 2 | aeState: 2
D/CameraCaptureCallback(17807): CameraCaptureCallback | state: STATE_WAITING_FOCUS | afState: 2 | aeState: 2
D/CameraCaptureCallback(17807): CameraCaptureCallback | state: STATE_WAITING_FOCUS | afState: 2 | aeState: 2
D/CameraCaptureCallback(17807): CameraCaptureCallback | state: STATE_WAITING_FOCUS | afState: 2 | aeState: 2
D/CameraCaptureCallback(17807): CameraCaptureCallback | state: STATE_WAITING_FOCUS | afState: 2 | aeState: 2
D/CameraCaptureCallback(17807): CameraCaptureCallback | state: STATE_WAITING_FOCUS | afState: 2 | aeState: 2
D/CameraCaptureCallback(17807): CameraCaptureCallback | state: STATE_WAITING_FOCUS | afState: 2 | aeState: 2
D/CameraCaptureCallback(17807): CameraCaptureCallback | state: STATE_WAITING_FOCUS | afState: 2 | aeState: 2
D/CameraCaptureCallback(17807): CameraCaptureCallback | state: STATE_WAITING_FOCUS | afState: 2 | aeState: 2
D/CameraCaptureCallback(17807): CameraCaptureCallback | state: STATE_WAITING_FOCUS | afState: 2 | aeState: 2
D/CameraCaptureCallback(17807): CameraCaptureCallback | state: STATE_WAITING_FOCUS | afState: 2 | aeState: 2
D/CameraCaptureCallback(17807): CameraCaptureCallback | state: STATE_WAITING_FOCUS | afState: 2 | aeState: 2
D/CameraCaptureCallback(17807): CameraCaptureCallback | state: STATE_WAITING_FOCUS | afState: 2 | aeState: 2
D/CameraCaptureCallback(17807): CameraCaptureCallback | state: STATE_WAITING_FOCUS | afState: 2 | aeState: 2
D/CameraCaptureCallback(17807): CameraCaptureCallback | state: STATE_WAITING_FOCUS | afState: 2 | aeState: 2
D/CameraCaptureCallback(17807): CameraCaptureCallback | state: STATE_WAITING_FOCUS | afState: 2 | aeState: 2
D/CameraCaptureCallback(17807): CameraCaptureCallback | state: STATE_WAITING_FOCUS | afState: 2 | aeState: 2
D/CameraCaptureCallback(17807): CameraCaptureCallback | state: STATE_WAITING_FOCUS | afState: 2 | aeState: 2
D/CameraCaptureCallback(17807): CameraCaptureCallback | state: STATE_WAITING_FOCUS | afState: 4 | aeState: 2
I/Camera  (17807): captureStillPicture
D/Camera  (17807): Updating builder with feature: ExposureLockFeature
D/Camera  (17807): Updating builder with feature: ExposurePointFeature
D/Camera  (17807): Updating builder with feature: ZoomLevelFeature
D/Camera  (17807): Updating builder with feature: AutoFocusFeature
D/Camera  (17807): Updating builder with feature: NoiseReductionFeature
I/Camera  (17807): updateNoiseReduction | currentSetting: fast
D/Camera  (17807): Updating builder with feature: FocusPointFeature
D/Camera  (17807): Updating builder with feature: ResolutionFeature
D/Camera  (17807): Updating builder with feature: SensorOrientationFeature
D/Camera  (17807): Updating builder with feature: FlashFeature
D/Camera  (17807): Updating builder with feature: ExposureOffsetFeature
D/Camera  (17807): Updating builder with feature: FpsRangeFeature
I/Camera  (17807): sending capture request
D/CameraCaptureCallback(17807): CameraCaptureCallback | state: STATE_CAPTURING | afState: 2 | aeState: 2
D/CameraCaptureCallback(17807): CameraCaptureCallback | state: STATE_CAPTURING | afState: 4 | aeState: 2
D/CameraCaptureCallback(17807): CameraCaptureCallback | state: STATE_CAPTURING | afState: 2 | aeState: 2
D/CameraCaptureCallback(17807): CameraCaptureCallback | state: STATE_CAPTURING | afState: 4 | aeState: 2
D/CameraCaptureCallback(17807): CameraCaptureCallback | state: STATE_CAPTURING | afState: 2 | aeState: 2
D/CameraCaptureCallback(17807): CameraCaptureCallback | state: STATE_CAPTURING | afState: 4 | aeState: 2
D/CameraCaptureCallback(17807): CameraCaptureCallback | state: STATE_CAPTURING | afState: 4 | aeState: 2
D/CameraCaptureCallback(17807): CameraCaptureCallback | state: STATE_CAPTURING | afState: 4 | aeState: 2
I/Camera  (17807): unlockAutoFocus
I/Camera  (17807): refreshPreviewCaptureSession
D/CameraCaptureCallback(17807): CameraCaptureCallback | state: STATE_CAPTURING | afState: 2 | aeState: 2
I/Camera  (17807): onImageAvailable
D/qdgralloc(17807): GetYUVPlaneInfo: Invalid format passed: 0x21
D/MIUIInput(17807): [KeyEvent] ViewRootImpl KeyEvent { action=ACTION_DOWN, keyCode=KEYCODE_BACK, scanCode=0, metaState=0, flags=0x48, repeatCount=0, eventTime=84250893, downTime=84250893, deviceId=-1, source=0x101, displayId=0 }
D/MIUIInput(17807): [KeyEvent] ViewRootImpl KeyEvent { action=ACTION_UP, keyCode=KEYCODE_BACK, scanCode=0, metaState=0, flags=0x48, repeatCount=0, eventTime=84250978, downTime=84250893, deviceId=-1, source=0x101, displayId=0 }
D/MIUIInput(17807): [MotionEvent] ViewRootImpl { action=ACTION_DOWN, id[0]=0, pointerCount=1, eventTime=84253037, downTime=84253037 } moveCount:0
D/MIUIInput(17807): [MotionEvent] ViewRootImpl { action=ACTION_UP, id[0]=0, pointerCount=1, eventTime=84253179, downTime=84253037 } moveCount:0
I/Camera  (17807): runPictureAutoFocus
I/Camera  (17807): lockAutoFocus
D/CameraCaptureCallback(17807): CameraCaptureCallback | state: STATE_WAITING_FOCUS | afState: 2 | aeState: 2
D/CameraCaptureCallback(17807): CameraCaptureCallback | state: STATE_WAITING_FOCUS | afState: 2 | aeState: 1
D/CameraCaptureCallback(17807): CameraCaptureCallback | state: STATE_WAITING_FOCUS | afState: 2 | aeState: 2
D/CameraCaptureCallback(17807): CameraCaptureCallback | state: STATE_WAITING_FOCUS | afState: 2 | aeState: 2
D/CameraCaptureCallback(17807): CameraCaptureCallback | state: STATE_WAITING_FOCUS | afState: 2 | aeState: 2
D/CameraCaptureCallback(17807): CameraCaptureCallback | state: STATE_WAITING_FOCUS | afState: 2 | aeState: 2
D/CameraCaptureCallback(17807): CameraCaptureCallback | state: STATE_WAITING_FOCUS | afState: 2 | aeState: 2
D/CameraCaptureCallback(17807): CameraCaptureCallback | state: STATE_WAITING_FOCUS | afState: 4 | aeState: 2
I/Camera  (17807): captureStillPicture
D/Camera  (17807): Updating builder with feature: ExposureLockFeature
D/Camera  (17807): Updating builder with feature: ExposurePointFeature
D/Camera  (17807): Updating builder with feature: ZoomLevelFeature
D/Camera  (17807): Updating builder with feature: AutoFocusFeature
D/Camera  (17807): Updating builder with feature: NoiseReductionFeature
I/Camera  (17807): updateNoiseReduction | currentSetting: fast
D/Camera  (17807): Updating builder with feature: FocusPointFeature
D/Camera  (17807): Updating builder with feature: ResolutionFeature
D/Camera  (17807): Updating builder with feature: SensorOrientationFeature
D/Camera  (17807): Updating builder with feature: FlashFeature
D/Camera  (17807): Updating builder with feature: ExposureOffsetFeature
D/Camera  (17807): Updating builder with feature: FpsRangeFeature
I/Camera  (17807): sending capture request
D/CameraCaptureCallback(17807): CameraCaptureCallback | state: STATE_CAPTURING | afState: 2 | aeState: 2
D/CameraCaptureCallback(17807): CameraCaptureCallback | state: STATE_CAPTURING | afState: 4 | aeState: 2
D/CameraCaptureCallback(17807): CameraCaptureCallback | state: STATE_CAPTURING | afState: 2 | aeState: 2
D/CameraCaptureCallback(17807): CameraCaptureCallback | state: STATE_CAPTURING | afState: 4 | aeState: 2
D/CameraCaptureCallback(17807): CameraCaptureCallback | state: STATE_CAPTURING | afState: 4 | aeState: 2
D/CameraCaptureCallback(17807): CameraCaptureCallback | state: STATE_CAPTURING | afState: 4 | aeState: 2
D/CameraCaptureCallback(17807): CameraCaptureCallback | state: STATE_CAPTURING | afState: 4 | aeState: 2
I/Camera  (17807): unlockAutoFocus
I/Camera  (17807): refreshPreviewCaptureSession
D/CameraCaptureCallback(17807): CameraCaptureCallback | state: STATE_CAPTURING | afState: 2 | aeState: 2
I/Camera  (17807): onImageAvailable
D/qdgralloc(17807): GetYUVPlaneInfo: Invalid format passed: 0x21
D/MIUIInput(17807): [KeyEvent] ViewRootImpl KeyEvent { action=ACTION_DOWN, keyCode=KEYCODE_BACK, scanCode=0, metaState=0, flags=0x48, repeatCount=0, eventTime=84256394, downTime=84256394, deviceId=-1, source=0x101, displayId=0 }
D/MIUIInput(17807): [KeyEvent] ViewRootImpl KeyEvent { action=ACTION_UP, keyCode=KEYCODE_BACK, scanCode=0, metaState=0, flags=0x48, repeatCount=0, eventTime=84256502, downTime=84256394, deviceId=-1, source=0x101, displayId=0 }
Lost connection to device.
Exited

```
</details> 
<details>
<summary>flutter doctor -v (mac)</summary>

```
[✓] Flutter (Channel stable, 3.0.5, on macOS 12.4 21F79 darwin-arm, locale en-IN)
    • Flutter version 3.0.5 at /Users/mahesh/Documents/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision f1875d570e (5 days ago), 2022-07-13 11:24:16 -0700
    • Engine revision e85ea0e79c
    • Dart version 2.17.6
    • DevTools version 2.12.2

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.0-rc4)
    • Android SDK at /Users/mahesh/Library/Android/sdk
    • Platform android-32, build-tools 33.0.0-rc4
    • ANDROID_HOME = /Users/mahesh/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 13.2.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)

[✓] IntelliJ IDEA Community Edition (version 2021.2.1)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 61.2.4
    • Dart plugin version 212.5080.8

[✓] VS Code (version 1.67.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.44.0

[✓] Connected device (3 available)
    • sdk gphone arm64 (mobile) • emulator-5554 • android-arm64  • Android 11 (API 30) (emulator)
    • macOS (desktop)           • macos         • darwin-arm64   • macOS 12.4 21F79 darwin-arm
    • Chrome (web)              • chrome        • web-javascript • Google Chrome 103.0.5060.114

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
```
</details>

**Comment by wwwqyhme** [comment]
@maheshmnj 

I'm tested on `黑鲨` which also belongs to Xiaomi but my android version is 12.

This is usually happens after I restarted the
application several times ,a `RejectedExecutionException` exception may throws in some times

``` 
java.util.concurrent.RejectedExecutionException: Task
android.hardware.camera2.impl.CallbackProxies$SessionStateCallbackProxy$$ExternalSyntheticLambda2@e125651 rejected from
java.util.concurrent.ThreadPoolExecutor@7c19eb6[Terminated, pool size = 0, active threads = 0, queued tasks = 0, completed tasks = 3]  
[        ] E/System  ( 5367):   at java.util.concurrent.ThreadPoolExecutor$AbortPolicy.rejectedExecution(ThreadPoolExecutor.java:2086) 
[        ] E/System  ( 5367):   at java.util.concurrent.ThreadPoolExecutor.reject(ThreadPoolExecutor.java:848)
[        ] E/System  ( 5367):   at java.util.concurrent.ThreadPoolExecutor.execute(ThreadPoolExecutor.java:1394)
[        ] E/System  ( 5367):   at java.util.concurrent.Executors$DelegatedExecutorService.execute(Executors.java:630)
[        ] E/System  ( 5367):   at
android.hardware.camera2.impl.CallbackProxies$SessionStateCallbackProxy.onClosed(CallbackProxies.java:104)
[        ] E/System  ( 5367):   at android.hardware.camera2.impl.CameraCaptureSessionImpl.close(CameraCaptureSessionImpl.java:596)     
[        ] E/System  ( 5367):   at android.hardware.camera2.impl.CameraCaptureSessionImpl.finalize(CameraCaptureSessionImpl.java:887)  
[        ] E/System  ( 5367):   at java.lang.Daemons$FinalizerDaemon.doFinalize(Daemons.java:291)
[        ] E/System  ( 5367):   at java.lang.Daemons$FinalizerDaemon.runInternal(Daemons.java:278)
[        ] E/System  ( 5367):   at java.lang.Daemons$Daemon.run(Daemons.java:139)
[        ] E/System  ( 5367):   at java.lang.Thread.run(Thread.java:920)
```

**Comment by wwwqyhme** [comment]
@maheshmnj  maybe this is a duplicate of https://github.com/flutter/flutter/issues/92973 ?

**Comment by wwwqyhme** [comment]
@maheshmnj 

I created a timer to start and stop image stream

``` dart
 _initializeControllerFuture = _controller.initialize().then((value) {
      timer = Timer.periodic(const Duration(seconds: 5), (timer) {
        if (_controller.value.isStreamingImages) {
          _controller.stopImageStream();
        } else {
          _controller.startImageStream((image) {});
        }
      });
    });
```

hope this can help you to reproduce  it

**Comment by wwwqyhme** [comment]
after I enabled strict mode of android , I got exception below:

tested on emulator(api 29)  and real device(android 12)

``` 
[ +435 ms] D/StrictMode(10298): StrictMode policy violation: android.os.strictmode.LeakedClosableViolation: A resource was acquired at
attached stack trace but never released. See java.io.Closeable for information on avoiding resource leaks.
[        ] D/StrictMode(10298):         at android.os.StrictMode$AndroidCloseGuardReporter.report(StrictMode.java:1877)
[        ] D/StrictMode(10298):         at dalvik.system.CloseGuard.warnIfOpen(CloseGuard.java:286)
[        ] D/StrictMode(10298):         at android.view.Surface.finalize(Surface.java:240)
[        ] D/StrictMode(10298):         at java.lang.Daemons$FinalizerDaemon.doFinalize(Daemons.java:289)
[        ] D/StrictMode(10298):         at java.lang.Daemons$FinalizerDaemon.runInternal(Daemons.java:276)
[        ] D/StrictMode(10298):         at java.lang.Daemons$Daemon.run(Daemons.java:137)
[        ] D/StrictMode(10298):         at java.lang.Thread.run(Thread.java:919)
[        ] D/StrictMode(10298): Caused by: java.lang.Throwable: Explicit termination method 'release' not called
[        ] D/StrictMode(10298):         at dalvik.system.CloseGuard.open(CloseGuard.java:237)
[        ] D/StrictMode(10298):         at android.view.Surface.setNativeObjectLocked(Surface.java:648)
[        ] D/StrictMode(10298):         at android.view.Surface.<init>(Surface.java:224)
[        ] D/StrictMode(10298):         at io.flutter.plugins.camera.Camera.createCaptureSession(Camera.java:296)
[        ] D/StrictMode(10298):         at io.flutter.plugins.camera.Camera.createCaptureSession(Camera.java:281)
[        ] D/StrictMode(10298):         at io.flutter.plugins.camera.Camera.startPreviewWithImageStream(Camera.java:1099)
[        ] D/StrictMode(10298):         at io.flutter.plugins.camera.MethodCallHandlerImpl.onMethodCall(MethodCallHandlerImpl.java:251)
[        ] D/StrictMode(10298):         at io.flutter.plugin.common.MethodChannel$IncomingMethodCallHandler.onMessage(MethodChannel.java:262)
[        ] D/StrictMode(10298):         at io.flutter.embedding.engine.dart.DartMessenger.invokeHandler(DartMessenger.java:295)        
[        ] D/StrictMode(10298):         at
io.flutter.embedding.engine.dart.DartMessenger.lambda$dispatchMessageToQueue$0$io-flutter-embedding-engine-dart-DartMessenger(DartMesse
nger.java:319)
[        ] D/StrictMode(10298):         at io.flutter.embedding.engine.dart.DartMessenger$$ExternalSyntheticLambda0.run(Unknown Source:12)
[        ] D/StrictMode(10298):         at android.os.Handler.handleCallback(Handler.java:883)
[        ] D/StrictMode(10298):         at android.os.Handler.dispatchMessage(Handler.java:100)
[        ] D/StrictMode(10298):         at android.os.Looper.loop(Looper.java:214)
[        ] D/StrictMode(10298):         at android.app.ActivityThread.main(ActivityThread.java:7356)
[        ] D/StrictMode(10298):         at java.lang.reflect.Method.invoke(Native Method)
[        ] D/StrictMode(10298):         at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:492)
[        ] D/StrictMode(10298):         at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:930)
[        ] W/System.err(10298): StrictMode VmPolicy violation with POLICY_DEATH; shutting down.
[        ] I/Process (10298): Sending signal. PID: 10298 SIG: 9
```

**Comment by danPyk** [comment]
I have the same warning followed with RejectedExecutionException as I'm describet here https://github.com/flutter/flutter/issues/90116#issuecomment-1205454826

**Comment by wwwqyhme** [comment]
I've looked the source code of `Surface`,this exception will only be happened in it's `finalize` method if we don't call release manually.

Due to `Surface` will release itself in  `finalize` method , so I don't think this will cause a memory problem, but the warning is annoying

**Comment by huycozy** [comment]
Hi @wwwqyhme 
I seem to have been able to reproduce the above warning even with the [Android native camera sample code](https://github.com/android/camera-samples/tree/main/Camera2Basic) as well. But it only happened once, the warning did not appear again after that. 
Can you try with [Android native camera sample code](https://github.com/android/camera-samples/tree/main/Camera2Basic)?

>after I enabled strict mode of android , I got exception below:
 
Did you enable `Strict mode` in Developer options?

**Comment by wwwqyhme** [comment]
I'm glad to try with  [Android native camera sample code](https://github.com/android/camera-samples/tree/main/Camera2Basic),but I don't think this issue is releated to that example.

They create a new Surface at [here](https://github.com/flutter/plugins/blob/c3da2d1dfc3e4bf4647091d3c785b1d9b3b16385/packages/camera/camera_android/android/src/main/java/io/flutter/plugins/camera/Camera.java#L407) ,but never call release on it

**Comment by wwwqyhme** [comment]
@huycozy 

I can see the warning `A resource failed to call release` , but only once after application started.


I override the onCreate method of  `CameraActivity`  and restart the application

``` kotlin
  override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        StrictMode.setThreadPolicy(
            ThreadPolicy.Builder()
                .detectNetwork() // or .detectAll() for all detectable problems
                .penaltyLog()
                .build()
        )
        StrictMode.setVmPolicy(
            VmPolicy.Builder()
                .detectLeakedClosableObjects()
                .penaltyLog()
                .penaltyDeath()
                .build()
        )
        activityCameraBinding = ActivityCameraBinding.inflate(layoutInflater)
        setContentView(activityCameraBinding.root)
    }
```

i got an exception :

``` 
2022-08-05 17:28:18.215 10053-10065/? D/StrictMode: StrictMode policy violation: android.os.strictmode.LeakedClosableViolation: A resource was acquired at attached stack trace but never released. See java.io.Closeable for information on avoiding resource leaks. Callsite: close
        at android.os.StrictMode$AndroidCloseGuardReporter.report(StrictMode.java:1992)
        at dalvik.system.CloseGuard.warnIfOpen(CloseGuard.java:347)
        at sun.nio.fs.UnixSecureDirectoryStream.finalize(UnixSecureDirectoryStream.java:580)
        at java.lang.Daemons$FinalizerDaemon.doFinalize(Daemons.java:291)
        at java.lang.Daemons$FinalizerDaemon.runInternal(Daemons.java:278)
        at java.lang.Daemons$Daemon.run(Daemons.java:139)
        at java.lang.Thread.run(Thread.java:920)
2022-08-05 17:28:18.215 10053-10065/? W/System.err: StrictMode VmPolicy violation with POLICY_DEATH; shutting down.
2022-08-05 17:28:18.215 10053-10065/? I/Process: Sending signal. PID: 10053 SIG: 9
```

tested on Android 12 real device

**Comment by huycozy** [comment]
@wwwqyhme 
I can see the exception now, but it only occurred once during the test (although I restart app many times)

Device info: Pixel 3a (Android 12)

<details>
<summary>Logs</summary>

```bash
Restarted application in 1,360ms.
I/Camera  ( 4783): close
W/System  ( 4783): A resource failed to call release.
E/System  ( 4783): Uncaught exception thrown by finalizer
E/System  ( 4783): java.util.concurrent.RejectedExecutionException: Task android.hardware.camera2.impl.CallbackProxies$SessionStateCallbackProxy$$ExternalSyntheticLambda2@5b5a244 rejected from java.util.concurrent.ThreadPoolExecutor@a5eff2d[Terminated, pool size = 0, active threads = 0, queued tasks = 0, completed tasks = 3]
E/System  ( 4783): 	at java.util.concurrent.ThreadPoolExecutor$AbortPolicy.rejectedExecution(ThreadPoolExecutor.java:2086)
E/System  ( 4783): 	at java.util.concurrent.ThreadPoolExecutor.reject(ThreadPoolExecutor.java:848)
E/System  ( 4783): 	at java.util.concurrent.ThreadPoolExecutor.execute(ThreadPoolExecutor.java:1394)
E/System  ( 4783): 	at java.util.concurrent.Executors$DelegatedExecutorService.execute(Executors.java:630)
E/System  ( 4783): 	at android.hardware.camera2.impl.CallbackProxies$SessionStateCallbackProxy.onClosed(CallbackProxies.java:104)
E/System  ( 4783): 	at android.hardware.camera2.impl.CameraCaptureSessionImpl.close(CameraCaptureSessionImpl.java:588)
E/System  ( 4783): 	at android.hardware.camera2.impl.CameraCaptureSessionImpl.finalize(CameraCaptureSessionImpl.java:879)
E/System  ( 4783): 	at java.lang.Daemons$FinalizerDaemon.doFinalize(Daemons.java:291)
E/System  ( 4783): 	at java.lang.Daemons$FinalizerDaemon.runInternal(Daemons.java:278)
E/System  ( 4783): 	at java.lang.Daemons$Daemon.run(Daemons.java:139)
E/System  ( 4783): 	at java.lang.Thread.run(Thread.java:920)
W/System  ( 4783): A resource failed to call release.
E/System  ( 4783): Uncaught exception thrown by finalizer
E/System  ( 4783): java.util.concurrent.RejectedExecutionException: Task android.hardware.camera2.impl.CallbackProxies$SessionStateCallbackProxy$$ExternalSyntheticLambda2@ecdd1f3 rejected from java.util.concurrent.ThreadPoolExecutor@88c9eb0[Terminated, pool size = 0, active threads = 0, queued tasks = 0, completed tasks = 3]
E/System  ( 4783): 	at java.util.concurrent.ThreadPoolExecutor$AbortPolicy.rejectedExecution(ThreadPoolExecutor.java:2086)
E/System  ( 4783): 	at java.util.concurrent.ThreadPoolExecutor.reject(ThreadPoolExecutor.java:848)
E/System  ( 4783): 	at java.util.concurrent.ThreadPoolExecutor.execute(ThreadPoolExecutor.java:1394)
E/System  ( 4783): 	at java.util.concurrent.Executors$DelegatedExecutorService.execute(Executors.java:630)
E/System  ( 4783): 	at android.hardware.camera2.impl.CallbackProxies$SessionStateCallbackProxy.onClosed(CallbackProxies.java:104)
E/System  ( 4783): 	at android.hardware.camera2.impl.CameraCaptureSessionImpl.close(CameraCaptureSessionImpl.java:588)
E/System  ( 4783): 	at android.hardware.camera2.impl.CameraCaptureSessionImpl.finalize(CameraCaptureSessionImpl.java:879)
E/System  ( 4783): 	at java.lang.Daemons$FinalizerDaemon.doFinalize(Daemons.java:291)
E/System  ( 4783): 	at java.lang.Daemons$FinalizerDaemon.runInternal(Daemons.java:278)
E/System  ( 4783): 	at java.lang.Daemons$Daemon.run(Daemons.java:139)
E/System  ( 4783): 	at java.lang.Thread.run(Thread.java:920)
I/Camera  ( 4783): open | onClosed
W/Camera  ( 4783): The selected imageFormatGroup is not supported by Android. Defaulting to yuv420
I/Camera  ( 4783): startPreview
I/Camera  ( 4783): CameraCaptureSession onConfigured
I/Camera  ( 4783): Updating builder settings
D/Camera  ( 4783): Updating builder with feature: ExposureLockFeature
D/Camera  ( 4783): Updating builder with feature: ExposurePointFeature
D/Camera  ( 4783): Updating builder with feature: ZoomLevelFeature
D/Camera  ( 4783): Updating builder with feature: AutoFocusFeature
D/Camera  ( 4783): Updating builder with feature: NoiseReductionFeature
I/Camera  ( 4783): updateNoiseReduction | currentSetting: fast
D/Camera  ( 4783): Updating builder with feature: FocusPointFeature
D/Camera  ( 4783): Updating builder with feature: ResolutionFeature
D/Camera  ( 4783): Updating builder with feature: SensorOrientationFeature
D/Camera  ( 4783): Updating builder with feature: FlashFeature
D/Camera  ( 4783): Updating builder with feature: ExposureOffsetFeature
D/Camera  ( 4783): Updating builder with feature: FpsRangeFeature
I/Camera  ( 4783): refreshPreviewCaptureSession
```
</details>

<details>
<summary>flutter doctor -v</summary>

```bash
[✓] Flutter (Channel stable, 3.0.5, on macOS 12.2.1 21D62 darwin-x64, locale en-VN)
    • Flutter version 3.0.5 at /Users/huynq/Documents/GitHub/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision f1875d570e (9 hours ago), 2022-07-13 11:24:16 -0700
    • Engine revision e85ea0e79c
    • Dart version 2.17.6
    • DevTools version 2.12.2

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/huynq/Library/Android/sdk
    • Platform android-33, build-tools 31.0.0
    • ANDROID_HOME = /Users/huynq/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 13.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)

[✓] IntelliJ IDEA Community Edition (version 2020.3.3)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] IntelliJ IDEA Community Edition (version 2022.1.1)
    • IntelliJ at /Users/huynq/Library/Application Support/JetBrains/Toolbox/apps/IDEA-C/ch-0/221.5591.52/IntelliJ IDEA CE.app
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] VS Code (version 1.69.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.44.0

[✓] Connected device (3 available)
    • SM T225 (mobile) • R9JT3004VRJ • android-arm64  • Android 11 (API 30)
    • macOS (desktop)  • macos       • darwin-x64     • macOS 12.2.1 21D62 darwin-x64
    • Chrome (web)     • chrome      • web-javascript • Google Chrome 103.0.5060.114

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!

```

```bash
[✓] Flutter (Channel master, 3.1.0-0.0.pre.2102, on macOS 12.5 21G72 darwin-x64, locale en-VN)
    • Flutter version 3.1.0-0.0.pre.2102 on channel master at /Users/huynq/Documents/GitHub/flutter_master
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 26c9d47311 (37 minutes ago), 2022-08-04 22:52:11 -0400
    • Engine revision a478045176
    • Dart version 2.19.0 (build 2.19.0-64.0.dev)
    • DevTools version 2.16.0

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/huynq/Library/Android/sdk
    • Platform android-33, build-tools 31.0.0
    • ANDROID_HOME = /Users/huynq/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 13.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 13E113
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)

[✓] IntelliJ IDEA Community Edition (version 2020.3.3)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] IntelliJ IDEA Community Edition (version 2022.1.1)
    • IntelliJ at /Users/huynq/Library/Application Support/JetBrains/Toolbox/apps/IDEA-C/ch-0/221.5591.52/IntelliJ IDEA CE.app
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] VS Code (version 1.69.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.46.0

[✓] Connected device (3 available)
    • SM T225 (mobile) • R9JT3004VRJ • android-arm64  • Android 11 (API 30)
    • macOS (desktop)  • macos       • darwin-x64     • macOS 12.5 21G72 darwin-x64
    • Chrome (web)     • chrome      • web-javascript • Google Chrome 103.0.5060.134

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!

```
</details>

Labeling this issue for further insight/investigation from the team.

**Comment by sulthanalihsan** [comment]
Any update on this issue

**Comment by clycheng2015** [comment]
same to me!

error info:

/System  ( 7384): Uncaught exception thrown by finalizer
E/System  ( 7384): java.util.concurrent.RejectedExecutionException: Task android.hardware.camera2.impl.-$$Lambda$CallbackProxies$SessionStateCallbackProxy$9H0ZdANdMrdpoq2bfIL2l3DVsKk@289feb5 rejected from java.util.concurrent.ThreadPoolExecutor@55ac94a[Terminated, pool size = 0, active threads = 0, queued tasks = 0, completed tasks = 3]
E/System  ( 7384): 	at java.util.concurrent.ThreadPoolExecutor$AbortPolicy.rejectedExecution(ThreadPoolExecutor.java:2086)
E/System  ( 7384): 	at java.util.concurrent.ThreadPoolExecutor.reject(ThreadPoolExecutor.java:848)
E/System  ( 7384): 	at java.util.concurrent.ThreadPoolExecutor.execute(ThreadPoolExecutor.java:1394)
E/System  ( 7384): 	at java.util.concurrent.Executors$DelegatedExecutorService.execute(Executors.java:630)
E/System  ( 7384): 	at android.hardware.camera2.impl.CallbackProxies$SessionStateCallbackProxy.onClosed(CallbackProxies.java:104)
E/System  ( 7384): 	at android.hardware.camera2.impl.CameraCaptureSessionImpl.close(CameraCaptureSessionImpl.java:587)
E/System  ( 7384): 	at android.hardware.camera2.impl.CameraCaptureSessionImpl.finalize(CameraCaptureSessionImpl.java:878)
E/System  ( 7384): 	at java.lang.Daemons$FinalizerDaemon.doFinalize(Daemons.java:292)
E/System  ( 7384): 	at java.lang.Daemons$FinalizerDaemon.runInternal(Daemons.java:279)
E/System  ( 7384): 	at java.lang.Daemons$Daemon.run(Daemons.java:140)
E/System  ( 7384): 	at java.lang.Thread.run(Thread.java:923)

- [flutter ]Flutter 2.10.0 • channel stable • https://github.com/flutter/flutter.git
Framework • revision 5f105a6ca7 (8 months ago) • 2022-02-01 14:15:42 -0800
Engine • revision 776efd2034
Tools • Dart 2.16.0 • DevTools 2.9.2

[✓] Flutter (Channel stable, 2.10.0, on macOS 12.3 21E230 darwin-arm, locale zh-Hans-CN)
    • Flutter version 2.10.0 at /Users/sanyu/.fvm/versions/2.10.0
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 5f105a6ca7 (8 months ago), 2022-02-01 14:15:42 -0800
    • Engine revision 776efd2034
    • Dart version 2.16.0
    • DevTools version 2.9.2
    • Pub download mirror https://pub.flutter-io.cn
    • Flutter download mirror https://storage.flutter-io.cn

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.0)
    • Android SDK at /Users/sanyu/Library/Android/sdk
    • Platform android-33, build-tools 33.0.0
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 13.4.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)

[✓] Connected device (2 available)
    • 21121119SC (mobile) • 899338810510 • android-arm64  • Android 11 (API 30)
    • Chrome (web)        • chrome       • web-javascript • Google

**Comment by MartinNiederl** [comment]
Same issue with flutter `3.4.0-35.0.pre.20` and camera plugin `0.10.0+3`

<details>
  <summary>flutter doctor --verbose</summary>

```
[√] Flutter (Channel master, 3.4.0-35.0.pre.20, on Microsoft Windows [Version 10.0.22000.918], locale de-AT)
    • Flutter version 3.4.0-35.0.pre.20 on channel master at D:\ProgramData\flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision c8cb0d9266 (2 days ago), 2022-10-01 02:09:28 -0400
    • Engine revision 2a58774add
    • Dart version 2.19.0 (build 2.19.0-264.0.dev)
    • DevTools version 2.17.0

[√] Windows Version (Installed version of Windows is version 10 or higher)

[√] Android toolchain - develop for Android devices (Android SDK version 33.0.0)
    • Android SDK at D:\ProgramData\Android\Sdk
    • Platform android-31, build-tools 33.0.0
    • Java binary at:
      C:\Users\marti\AppData\Local\JetBrains\Toolbox\apps\AndroidStudio\ch-0\213.7172.25.2113.9014738\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files (x86)\Google\Chrome\Application\chrome.exe

[X] Visual Studio - develop for Windows
    X Visual Studio not installed; this is necessary for Windows development.
      Download at https://visualstudio.microsoft.com/downloads/.
      Please install the "Desktop development with C++" workload, including all of its default components

[√] Android Studio (version 2021.3)
    • Android Studio at C:\Users\marti\AppData\Local\JetBrains\Toolbox\apps\AndroidStudio\ch-0\213.7172.25.2113.9014738
    • Flutter plugin version 70.0.3
    • Dart plugin version 213.7433
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)

[√] IntelliJ IDEA Ultimate Edition (version 2022.2)
    • IntelliJ at C:\Users\marti\AppData\Local\JetBrains\Toolbox\apps\IDEA-U\ch-0\222.4167.29
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart

[√] VS Code, 64-bit edition (version 1.71.2)
    • VS Code at C:\Program Files\Microsoft VS Code
    • Flutter extension version 3.48.0

[√] Connected device (4 available)
    • SM A500FU (mobile) • 951ad3cd • android-arm    • Android 10 (API 29)
    • Windows (desktop)  • windows  • windows-x64    • Microsoft Windows [Version 10.0.22000.918]
    • Chrome (web)       • chrome   • web-javascript • Google Chrome 105.0.5195.127
    • Edge (web)         • edge     • web-javascript • Microsoft Edge 105.0.1343.53

[√] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 1 category.
```
</details>

<details>
  <summary>Logs</summary>

```
W/System  (31499): A resource failed to call release. 
E/System  (31499): Uncaught exception thrown by finalizer
E/System  (31499): java.util.concurrent.RejectedExecutionException: Task android.hardware.camera2.impl.-$$Lambda$CallbackProxies$SessionStateCallbackProxy$9H0ZdANdMrdpoq2bfIL2l3DVsKk@468e726 rejected from java.util.concurrent.ThreadPoolExecutor@47c2f67[Terminated, pool size = 0, active threads = 0, queued tasks = 0, completed tasks = 3]
E/System  (31499): 	at java.util.concurrent.ThreadPoolExecutor$AbortPolicy.rejectedExecution(ThreadPoolExecutor.java:2086)
E/System  (31499): 	at java.util.concurrent.ThreadPoolExecutor.reject(ThreadPoolExecutor.java:848)
E/System  (31499): 	at java.util.concurrent.ThreadPoolExecutor.execute(ThreadPoolExecutor.java:1394)
E/System  (31499): 	at java.util.concurrent.Executors$DelegatedExecutorService.execute(Executors.java:630)
E/System  (31499): 	at android.hardware.camera2.impl.CallbackProxies$SessionStateCallbackProxy.onClosed(CallbackProxies.java:104)
E/System  (31499): 	at android.hardware.camera2.impl.CameraCaptureSessionImpl.close(CameraCaptureSessionImpl.java:535)
E/System  (31499): 	at android.hardware.camera2.impl.CameraCaptureSessionImpl.finalize(CameraCaptureSessionImpl.java:826)
E/System  (31499): 	at java.lang.Daemons$FinalizerDaemon.doFinalize(Daemons.java:289)
E/System  (31499): 	at java.lang.Daemons$FinalizerDaemon.runInternal(Daemons.java:276)
E/System  (31499): 	at java.lang.Daemons$Daemon.run(Daemons.java:137)
E/System  (31499): 	at java.lang.Thread.run(Thread.java:919)
```
</details>

**Comment by CaoGiaHieu-dev** [comment]
Any update on this issue?

**Comment by tigrenok00** [comment]
Any updates on this? I experience a lot of lagging on the UI when stopping and starting image stream a few times

**Comment by alec-c4** [comment]
Hey, any update?

**Comment by goerlitz** [comment]
Any update?
I see this issue while using a barcode scanner und switching between Camera/Scanner and other app pages.

**Comment by bakboem** [comment]
I also .. Any update?
`E/System  (  431): Uncaught exception thrown by finalizer
E/System  (  431): java.util.concurrent.RejectedExecutionException: Task android.hardware.camera2.impl.-$$Lambda$CallbackProxies$SessionStateCallbackProxy$9H0ZdANdMrdpoq2bfIL2l3DVsKk@a3d615a rejected from java.util.concurrent.ThreadPoolExecutor@5680f8b[Terminated, pool size = 0, active threads = 0, queued tasks = 0, completed tasks = 3]
E/System  (  431):      at java.util.concurrent.ThreadPoolExecutor$AbortPolicy.rejectedExecution(ThreadPoolExecutor.java:2086)
E/System  (  431):      at java.util.concurrent.ThreadPoolExecutor.reject(ThreadPoolExecutor.java:848)
E/System  (  431):      at java.util.concurrent.ThreadPoolExecutor.execute(ThreadPoolExecutor.java:1394)
E/System  (  431):      at java.util.concurrent.Executors$DelegatedExecutorService.execute(Executors.java:630)
E/System  (  431):      at android.hardware.camera2.impl.CallbackProxies$SessionStateCallbackProxy.onClosed(CallbackProxies.java:104)
E/System  (  431):      at android.hardware.camera2.impl.CameraCaptureSessionImpl.close(CameraCaptureSessionImpl.java:535)
E/System  (  431):      at android.hardware.camera2.impl.CameraCaptureSessionImpl.finalize(CameraCaptureSessionImpl.java:826)
E/System  (  431):      at java.lang.Daemons$FinalizerDaemon.doFinalize(Daemons.java:289)
E/System  (  431):      at java.lang.Daemons$FinalizerDaemon.runInternal(Daemons.java:276)
E/System  (  431):      at java.lang.Daemons$Daemon.run(Daemons.java:137)
E/System  (  431):      at java.lang.Thread.run(Thread.java:919)`

**Comment by camsim99** [comment]
This issue should be fixed in the [CameraX implementation](https://pub.dev/packages/camera_android_camerax) of the camera plugin, so please try it out!

If you find a similar issue with that implementation (or any other issues), please feel free to file a separate issue.
